tests/follow-data-only/copydb.sh

```diff
@@ -20,10 +20,21 @@ psql -d ${PGCOPYDB_TARGET_PGURI} -f /usr/src/pgcopydb/ddl.sql
 # insert a first batch of 10 rows (1..10)
 psql -v a=1 -v b=10 -d ${PGCOPYDB_SOURCE_PGURI} -f /usr/src/pgcopydb/dml.sql
 
+# run concurrent transactions in background and take snapshot in between to
+# verify we copy newly inserted data when the snapshot is taken
+bash ./run-background-traffic.sh &
+BACKGROUND_TRAFFIC_PID=$!
```

Comment:
Is there a way to move the background traffic to the inject service that runs in the background already?

Reply:
I want to have on-going traffic for at least some time before we take the snapshot (2 seconds here), that can be easily achieved here IMO. What do you think?

Reply:
I see. Then let's add a comment that's what is happening, and that having control over the sequence of events is not a good fit for multiple services. 

Also, can we use the bash `coproc` facility more than once? I believe we can give each coproc a name, see https://www.gnu.org/software/bash/manual/html_node/Coprocesses.html. Would be best to use the same background job mechanism twice rather than two different ways to do the same thing.

Reply:
I tried `coproc` at first, but it was giving this error,
```
warning: execute_coproc: coproc still exists
```
on a quick search, it seems we can't have a parallel `coproc` without closing an existing one.

Reply:
You need to use the coproc syntax with a NAME, you can't run multiple coproc with the same name.

Reply:
Yes, I gave them different unique names. Let me see if I can still reproduce that.

Reply:
Oh it worked but it still gave me this warning
```
/usr/src/pgcopydb/copydb.sh: line 31: warning: execute_coproc: coproc [35:BACKGROUND_TRAFFIC] still exists
```

here are the changes
```diff
diff --git a/tests/follow-data-only/copydb.sh b/tests/follow-data-only/copydb.sh
index 594e699..fc340b7 100755
--- a/tests/follow-data-only/copydb.sh
+++ b/tests/follow-data-only/copydb.sh
@@ -22,14 +22,13 @@ psql -v a=1 -v b=10 -d ${PGCOPYDB_SOURCE_PGURI} -f /usr/src/pgcopydb/dml.sql

 # run concurrent transactions in background and take snapshot in between to
 # verify we copy newly inserted data when the snapshot is taken
-bash ./run-background-traffic.sh &
-BACKGROUND_TRAFFIC_PID=$!
+coproc BACKGROUND_TRAFFIC ( bash ./run-background-traffic.sh )

 # wait for few seconds to allow snapshot to happen in between the traffic
 sleep 2

 # grab a snapshot on the source database
-coproc ( pgcopydb snapshot --follow --plugin wal2json --notice )
+coproc SNAPSHOT ( pgcopydb snapshot --follow --plugin wal2json --notice )

 sleep 2

@@ -51,8 +50,8 @@ pgcopydb follow --notice
 # the follow command ends when reaching endpos, set in inject.sh
 pgcopydb stream cleanup

-kill -TERM ${COPROC_PID}
-wait ${COPROC_PID}
+kill -TERM ${SNAPSHOT_PID}
+wait ${SNAPSHOT_PID}

 # check how many rows we have on source and target
 sql="select count(*), sum(some_field) from table_a"
```

Reply:
> Then let's add a comment that's what is happening

I have already added some comments, I think they are enough, can you please suggest what are we missing?

> and that having control over the sequence of events is not a good fit for multiple services.

Can you please elaborate?

Reply:
Okay Sorry about that, I am now reading more about coproc and bash seems to only guarantee coproc for a single sub-process at a time. Let's keep your initial implementation. About the comment, here is my suggestion:

``` diff
- # run concurrent transactions in background and take snapshot in between to
- # verify we copy newly inserted data when the snapshot is taken
+ # take a snapshot with concurrent activity happening
+ # it would be hard to sync concurrent activity in the inject service, so use a job instead
```

Reply:
I have made the changes, please reivew.